### PR TITLE
Fix links in Dask cuDF documentation

### PR DESCRIPTION
**Pull request by rjzamora**
More follow-up fixes to the recent Dask-cuDF documentation additions.

### CODE REVIEW

File: docs/dask_cudf/source/best_practices.rst [hunk 1]
```diff
@@ -297,11 +297,16 @@ bottleneck is typically device-to-host memory spilling.
 Although every workflow is different, the following guidelines
 are often recommended:
 
-* `Use a distributed cluster with Dask-CUDA workers <Use Dask-CUDA>`_
-* `Use native cuDF spilling whenever possible <Enable cuDF Spilling>`_
+* Use a distributed cluster with `Dask-CUDA <https://docs.rapids.ai/api/dask-cuda/stable/>`__ workers
```
Comment (rjzamora) [inline docs/dask_cudf/source/best_practices.rst:300]: I'm honestly not sure how to properly link to a different section within this page, so I'm just linking to the dask-cuda docs, etc. Happy to revise if others have advice.

File: docs/dask_cudf/source/best_practices.rst [hunk 2]
```diff
@@ -297,11 +297,16 @@ bottleneck is typically device-to-host memory spilling.
...
+
+* Use native cuDF spilling whenever possible (`Dask-CUDA spilling documentation <https://docs.rapids.ai/api/dask-cuda/24.10/spilling/>`__)
+
 * Avoid shuffling whenever possible
-  * Use ``split_out=1`` for low-cardinality groupby aggregations
-  * Use ``broadcast=True`` for joins when at least one collection comprises a small number of partitions (e.g. ``<=5``)
+
+   * Use ``split_out=1`` for low-cardinality groupby aggregations
```
Comment (rjzamora) [inline docs/dask_cudf/source/best_practices.rst:306]: Not completely sure how to get a nested list to render "correctly"

**Review by galipremsagar — APPROVED** [review]
I pushed some changes that will not hardcode the version URLs. Approving as the docs links and list now look fine.

File: docs/dask_cudf/source/best_practices.rst (hunk 2 above)
Comment (galipremsagar) [reply docs/dask_cudf/source/best_practices.rst]: Fixed it.

File: docs/dask_cudf/source/best_practices.rst (hunk 1 above)
Comment (vyasr) [reply docs/dask_cudf/source/best_practices.rst:300]: You can do inter-page links with syntax like this: https://www.sphinx-doc.org/en/master/usage/referencing.html#cross-referencing-syntax. No need to wait for this right now though.

**Review by vyasr — APPROVED** [review]

File: docs/dask_cudf/source/best_practices.rst (hunk 1 above)
Comment (galipremsagar) [reply docs/dask_cudf/source/best_practices.rst:300]: I just did that for this page, we can do that more broadly in `24.12`

**Comment by rjzamora** [comment]
Thanks so much for your help with this @galipremsagar !

**Pull request merged**